driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java

```diff
@@ -735,4 +761,70 @@ private boolean references(DataType dataType, String typeName) {
             return false;
         }
     }
+
+    private static class V4SchemaParser extends V3SchemaParser {
+
+        private static final String SELECT_VIRTUAL_KEYSPACES = "SELECT * FROM system_virtual_schema.keyspaces";
+        private static final String SELECT_VIRTUAL_TABLES = "SELECT * FROM system_virtual_schema.tables";
+        private static final String SELECT_VIRTUAL_COLUMNS = "SELECT * FROM system_virtual_schema.columns";
+
+
+        private static final String TABLE_NAME = "table_name";
+
+        @Override
+        SystemRows fetchSystemRows(Cluster cluster, SchemaElement targetType, String targetKeyspace, String targetName, List<String> targetSignature, Connection connection, VersionNumber cassandraVersion)
+                throws ConnectionException, BusyConnectionException, ExecutionException, InterruptedException {
+
+            boolean isSchemaOrKeyspace = (targetType == null || targetType == KEYSPACE);
+
+            ResultSetFuture ksFuture = null,
+                    udtFuture = null,
+                    cfFuture = null,
+                    colsFuture = null,
+                    functionsFuture = null,
+                    aggregatesFuture = null,
+                    indexesFuture = null,
+                    viewsFuture = null,
+                    virtualKeyspacesFuture = null,
+                    virtualTableFuture = null,
+                    virtualColumnsFuture = null;
+
+            ProtocolVersion protocolVersion = cluster.getConfiguration().getProtocolOptions().getProtocolVersion();
+
+            if (isSchemaOrKeyspace)
+                ksFuture = queryAsync(SELECT_KEYSPACES + whereClause(targetType, targetKeyspace, targetName, targetSignature), connection, protocolVersion);
+                virtualKeyspacesFuture = queryAsync(SELECT_VIRTUAL_KEYSPACES + whereClause(targetType, targetKeyspace, targetName, targetSignature), connection, protocolVersion);
+                virtualColumnsFuture = queryAsync(SELECT_VIRTUAL_COLUMNS + whereClause(targetType, targetKeyspace, targetName, targetSignature), connection, protocolVersion);
+                virtualTableFuture = queryAsync(SELECT_VIRTUAL_TABLES + whereClause(targetType, targetKeyspace, targetName, targetSignature), connection, protocolVersion);
```

Comment:
You need curly braces around this block, currently only the first statement is executed conditionally.

Reply:
Good catch